twistlock/datadog_checks/twistlock/twistlock.py

```diff
@@ -0,0 +1,322 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from collections import Counter
+import time
+from datetime import datetime, timedelta
+
+from six import iteritems
+
+import requests
+
+from datadog_checks.base import AgentCheck
+
+from .config import Config
+
+REGISTRY_SCAN_DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+SCAN_DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+LICENCE_DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+DOCKERIO_PREFIX = "docker.io/"
+
+# min_severity tag allows to query number of vulns for a severity target easily
+SEVERITY_TAGS = {
+    "low": ["severity:low", "min_severity:low"],
+    "medium": ["severity:medium", "min_severity:low", "min_severity:medium"],
+    "high": ["severity:high", "min_severity:low", "min_severity:medium", "min_severity:high"],
+    "critical": ["severity:critical", "min_severity:low", "min_severity:medium",
+                 "min_severity:high", "min_severity:critical"]
+}
+
+
+class TwistlockCheck(AgentCheck):
+    NAMESPACE = 'twistlock'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+
+        msg = 'This check has not run before, '
+        msg += 'it will send all the new vulnerabilities that happened in the past day'
+        self.log.debug(msg)
+        self.last_run = datetime.now() - timedelta(days=1)
+
+    def check(self, instance):
+        if 'url' not in instance:
+            raise Exception('Instance missing "url" value.')
+
+        self.config = Config(instance)
+
+        self.current_date = datetime.now()
+        self.warning_date = self.current_date - timedelta(hours=7)
+        self.critical_date = self.current_date - timedelta(days=1)
+
+        self.report_license_expiration()
+        self.report_registry_scan()
+        self.report_images_scan()
+        self.report_hosts_scan()
+        self.report_container_compliance()
+
+        self.report_vulnerabilities()
+
+        self.last_run = datetime.now()
+
+    def report_license_expiration(self):
+        service_check_name = self.NAMESPACE + ".license_ok"
+        try:
+            license = self._retrieve_json("/api/v1/settings/license")
+        except Exception as e:
+            self.warning("cannot retrieve license data: {}".format(e))
+            self.service_check(service_check_name, AgentCheck.CRITICAL, tags=self.config.tags)
+            raise e
+            return None
+
+        expiration_date = datetime.strptime(license.get("expiration_date"), LICENCE_DATE_FORMAT)
```

Comment:
nit: can we standardize on which regional spelling of `license` we use?

Reply:
did I use different spellings in the check?

Reply:
Oh I found it, haha I didn't notice that, it's from the part of the check that was written for internal use